lib/services/access_list.go

```diff
@@ -91,3 +93,87 @@ func UnmarshalAccessList(data []byte, opts ...MarshalOption) (*accesslist.Access
 	}
 	return accessList, nil
 }
+
+// IsOwner will return true if the user is an owner for the current list.
+func IsOwner(identity tlsca.Identity, accessList *accesslist.AccessList) error {
+	isOwner := false
+	for _, owner := range accessList.Spec.Owners {
+		if owner.Name == identity.Username {
+			isOwner = true
+			break
+		}
+	}
+
+	// An opaque access denied error.
+	accessDenied := trace.AccessDenied("access denied")
+
+	// User is not an owner, so we'll access denied.
+	if !isOwner {
+		return accessDenied
+	}
+
+	if !userMeetsRequirements(identity, accessList.Spec.OwnershipRequires) {
+		return accessDenied
+	}
```

Comment:
nit: I think we should avoid reusing the trace error for these if statements.

I think the idea here was to avoid repetition, but keep in mind that when the trace error is created, the stack trace will reference the line that it was created on, so it won't be possible to see from a stack trace whether the error comes from the first check or the second.

Reply:
unless that was intentional? Maybe it was based on "opque error" comment?

Reply:
Yes, exactly. The idea here is not to expose too much info.